lifinance/sdk
# fix: increment gas on retry if out of gas

# FEATURE

Retry with increased gas if transaction fails with out of gas error [LF-2862]

## TL;DR
Automatically retry transactions with increased gas if they fail with out of gas errors.

## DESCRIPTION
This feature automatically retries transactions with increased gas if they fail with out of gas errors. When attempting to send a transaction for bridging or swapping, if the initial gas is too low and the transaction fails with an out of gas error, the feature will automatically retry the transaction with increased gas to ensure that it is successfully processed.

## STEPS TO TEST
1. Reduce the gas to 21000 from Metamask before sending a transaction for bridging or swapping.
2. Observe that the transaction fails with an out of gas error.
3. The feature will automatically retry the transaction with increased gas.
4. Verify that the transaction is successfully processed.

## EXPECTED RESULTS
The feature should automatically retry transactions with increased gas if they fail with out of gas errors, and the transactions should be successfully processed.

## CHECKLIST
- [ ] I have commented the code I wrote to make sure the next developers can understand it.
- [ ] I have added tests that cover the new functionality.
- [ ] I have made sure that existing tests are still passing.

https://user-images.githubusercontent.com/35174288/236535777-e4a67fc2-ad13-4d41-9547-3fcd46431f22.mp4








[LF-2862]: https://lifi.atlassian.net/browse/LF-2862?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
